llvm/lib/Target/AMDGPU/AMDGPUHSAMetadataStreamer.cpp

```diff
@@ -409,6 +430,12 @@ void MetadataStreamerMsgPackV4::emitKernelArg(
       Arg[".is_pipe"] = Arg.getDocument()->getNode(true);
   }
 
+  // Add original argument index and offset to the metadata
+  if (OriginalArgIndex != ~0U) {
+    Arg[".original_arg_index"] = Arg.getDocument()->getNode(OriginalArgIndex);
```

Comment:
Honestly, I agree with @arsenm that this is indeed an ABI breaking change, even though the ABI change seems to be transparent to end users.